src/main/java/com/squareup/kotlinpoet/TypeVariableName.kt

```diff
@@ -27,13 +27,15 @@ import kotlin.reflect.KVariance
 
 class TypeVariableName private constructor(
   val name: String,
-  val bounds: List<TypeName>,
+  inputBounds: List<TypeName>,
   val variance: KModifier? = null,
   val reified: Boolean = false,
   nullable: Boolean = false,
   annotations: List<AnnotationSpec> = emptyList()
 ) : TypeName(nullable, annotations) {
 
+  val bounds = if (inputBounds.isEmpty()) listOf(NULLABLE_ANY) else inputBounds
```

Comment:
can you push this out to the calling code? As far as I know only `TypeVariableName.Companion#of` actually benefits from this behavior.

And if you push it into that method, you can just make `listOf(NULLABLE_ANY)` the default value of the parameter. While you’re there you could make variance default to null so calling this thing is super easy.

Reply:
can do. Originally did this because I saw that this consutructor was used in a few methods in TypeVariableName, but after looking again there's no danger of them passing empty lists

Reply:
Couple other questions

- should `variance` be moved to before the `bounds` arg? Feels weird to have a vararg in the middle of the params, and now they'd both have default values? Going to do that in my update but can change if needed
- what about the overloads that accept `Type` and `KClass`? Don't specify that arg if they're empty?